src/test.zig

```diff
@@ -785,108 +1010,99 @@ pub const TestContext = struct {
+                            name_prefix,
+                            @tagName(backend),
+                            try target.zigTriple(ctx.arena),
+                        });
+                        const next = ctx.cases.items.len;
+                        try ctx.cases.append(.{
+                            .name = name,
+                            .target = target,
+                            .backend = backend,
+                            .updates = std.ArrayList(TestContext.Update).init(ctx.cases.allocator),
+                            .is_test = is_test,
+                            .output_mode = output_mode,
+                            .link_libc = backend == .llvm,
+                            .files = std.ArrayList(TestContext.File).init(ctx.cases.allocator),
+                        });
+                        try cases.append(next);
+                    }
                 }
+            }
 
-                // All following lines are expected error messages
-                while (manifest_it.next()) |line| try errors.append(try ctx.arena.dupe(u8, std.mem.trim(u8, line[2..], " \t")));
-
-                const case = opt_case orelse case: {
-                    ctx.cases.append(TestContext.Case{
-                        .name = name,
-                        .target = .{},
-                        .backend = backend,
-                        .updates = std.ArrayList(TestContext.Update).init(ctx.cases.allocator),
-                        .is_test = is_test,
-                        .output_mode = output_mode,
-                        .files = std.ArrayList(TestContext.File).init(ctx.cases.allocator),
-                    }) catch @panic("out of memory");
-                    const case = &ctx.cases.items[ctx.cases.items.len - 1];
-                    opt_case = case;
-                    break :case case;
-                };
-                switch (strategy) {
-                    .independent => {
-                        case.name = case_name;
-                        case.addError(src, errors.items);
+            for (cases.items) |case_index| {
+                const case = &ctx.cases.items[case_index];
+                switch (manifest.@"type") {
+                    .@"error" => {
+                        const errors = try manifest.trailingAlloc(ctx.arena);
+                        switch (strategy) {
+                            .independent => {
+                                case.addError(src, errors);
+                            },
+                            .incremental => {
+                                case.addErrorNamed("update", src, errors);
+                            },
+                        }
                     },
-                    .incremental => {
-                        case.addErrorNamed(case_name, src, errors.items);
+                    .run => {
+                        var output = std.ArrayList(u8).init(ctx.arena);
```

Comment:
Does this create a new arena instance? Otherwise this looks like it would leak memory.
But maybe, I am just blind.